Add docs for non diverge branches with Packit (#931)
packit/packit.dev: docs/fedora-releases-guide/index.md

  - for **source archive(s) generation**, you can utilise e.g. `pre-sync` action, see 
  [this example](/docs/configuration/examples#custom-archive-creation-for-release-syncing)

#### Keep dist-git branches non-divergent

> If you want to preserve your dist-git branches from diverge,

Reviewer: ```suggestion
If you want to keep your dist-git branches from diverging,
```

> you can use the new `dist_git_branches` syntax.

Reviewer: ```suggestion
you can use the new `dist_git_branches` syntax:
```

  
```yaml
  dist_git_branches:
    rawhide:
      fast_forward_merge_into: [fedora-branched]
    epel-9: {}    
      
```

> In this example packit runs the downstream synchronization just for the
> `rawhide` and `epel` branches and opens a new pull request for every
> `fedora-branched` branch trying to fast forward the changes introduced
> in the `rawhide` branch.

Reviewer: ```suggestion
In this example, Packit runs the downstream synchronization just for the
`rawhide` and `epel-9` branches and opens a new pull request for every
`fedora-branched` branch trying to fast forward the changes introduced
in the `rawhide` branch.
```

> In this example, Packit runs the downstream synchronization just for the
> `rawhide` and `epel-9` branches and opens a new pull request for every
> `fedora-branched` branch trying to fast forward the changes introduced
> in the `rawhide` branch.

Reviewer 2: I am not sure if this is still easy to understand but I see this is really hard to describe simply...;)

```suggestion
In this example, Packit runs the downstream synchronization process for the
`rawhide` and `epel-9` branches as usual. But Packit also opens a new pull request reusing the commit from `rawhide` for every
`fedora-branched` branch so it can be fast-forwarded when merging.
```

> :::warning How to reconcile divergent branches

Reviewer 2: :+1:

> If you are already using Packit then your branches can have diverged.
> You need to reconcile them before using the new dist_git_branches 
> syntax.
> For the configuration example above and for the state of the branched
> Fedora releases as today, you need to

Reviewer: ```suggestion
If you are already using Packit then your branches can have diverged.
You need to reconcile them before using the new dist_git_branches 
syntax.
For the configuration example above and for the state of the branched
Fedora releases as today, you need to do:
```

```
checkout rawhide
merge f39
merge f40

> merge f41

Reviewer: shouldn't these all be starting with `git`?
